### ISSUE TITLE
Question about Drop Connect & Stochastic Depth

### ISSUE THREAD

**Issue by Bear-kai** [lukemelas/EfficientNet-PyTorch#104]
The parameter drop_connect_rate is used for stochastic depth of the network, but the function drop_connect() seems to drop samples ?

`def drop_connect(inputs, p, training):           
    """ Drop connect. """                        
    if not training: return inputs
    batch_size = inputs.shape[0]
    keep_prob = 1 - p
    random_tensor = keep_prob
    random_tensor += torch.rand([batch_size, 1, 1, 1], dtype=inputs.dtype, device=inputs.device)
    binary_tensor = torch.floor(random_tensor)
    output = inputs / keep_prob * binary_tensor
    return output`

Stochastic Depth should drop the building block randomly.
I don't understand the above function. Anyone can help ?!

**Comment by Bear-kai** [comment]
Ok, I find that the DropConnect (ICML 2013) is a generalization of Dropout. So this pytorch reimplementation  does not use the stochastic depth strategy.

**Comment by Renthal** [comment]
Relevant issue: https://github.com/tensorflow/tpu/issues/494#issuecomment-532968956

**Comment by JVGD** [comment]
Thank you for the clarification since I came here with similar questions. I still have some doubts with regard to the implementation. Specifically, I was expecting to see outputs being dropped out according to the `binary_tensor` (with multiplication):
```
output = inputs * binary_tensor
```
However I see that apart from the dropping (implemented with `binary_tensor`) there is also a scaling factor tensor (`keep_prob`) applied to the inputs:
```
output = inputs / keep_prob * binary_tensor
```
Why is that? I know there are kinds of Gaussian Dropouts where inputs are scaled up or down according to Normal distribution... but in here I'm not quite sure why.
